src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -197,6 +194,24 @@ class SharedLinkSection extends React.Component<Props, State> {
                     onCopyError();
                 });
         }
+
+        // if ESL ftux tooltip is showing on initial mount, we call onShow
+        const allowedPermissionLevels = this.getAllowedPermissionLevels(
+            canChangeAccessLevel,
+            permissionLevel,
+            isDownloadSettingAvailable,
+            isEditSettingAvailable,
+            isAllowEditSharedLinkForFileEnabled,
+        );
+
+        if (
```

Comment:
Since this is in `componentDidUpdate`, what is causing this to be shown only once?

Reply:
I originally had it in `componentDidMount` but then I noticed that the `allowedPermissionlevels` array that was coming back was empty. I then looked at `componentDidUpdate` and saw this comment that had been left:
```
    // We handle didUpdate but not didMount because
    // the component initially renders with empty data
    // in order to start showing UI components.
    // When getInitialData completes in the parent we
    // rerender with correct sharedLink data and can
    // check whether to auto create a new one.
    // Note: we are assuming the 2nd render is safe
    // to start doing this check.
```

So it seems that the first mount only renders with empty data so we aren't able to perform our check there.

Regarding your question about showing it only once, I'm wondering if there is some way to check to see if `onShow` has already been called for a specific FTUX. If there is, then we could apply that flag in here to make sure that we only call `onShow` once.

Reply:
Ok, I believe we can add a piece of state to determine if onShow has been called similar to the if statement above with isCopySuccessful